### PR TITLE
made image pull policy configurable

### PR DESCRIPTION
**Pull request by KavishShah09**

### CODE REVIEW

File: runner/utils/driver/docker/helpers.go [hunk 1]
```diff
@@ -20,45 +20,62 @@ func getServiceDomain(projectID, serviceID string) string {
...
 
 func (d *Docker) pullImageIfDoesntExists(ctx context.Context, projectID string, taskDocker model.Docker) error {
-	_, _, err := d.client.ImageInspectWithRaw(ctx, taskDocker.Image)
-	if err != nil {
-		// image doesn't exist locally
-		if taskDocker.Secret != "" {
-			data, err := ioutil.ReadFile(fmt.Sprintf("%s/%s/%s.json", d.secretPath, projectID, taskDocker.Secret))
+	if taskDocker.ImagePullPolicy == model.PullIfNotExists {
+		_, _, err := d.client.ImageInspectWithRaw(ctx, taskDocker.Image)
+		if err != nil {
```
Comment (YourTechBud) [inline runner/utils/driver/docker/helpers.go]: Simply return if error is nil. You are pulling the image down there anyways

File: runner/utils/driver/docker/helpers.go [hunk 2]
```diff
@@ -20,45 +20,62 @@ func getServiceDomain(projectID, serviceID string) string {
 }
 
 func (d *Docker) pullImageIfDoesntExists(ctx context.Context, projectID string, taskDocker model.Docker) error {
```
Comment (YourTechBud) [inline runner/utils/driver/docker/helpers.go]: Change the name of the function

File: runner/utils/driver/docker/helpers.go (hunk 1 above)
Comment (YourTechBud) [reply runner/utils/driver/docker/helpers.go]: You didn't do this @KavishShah09

Comment (KavishShah09) [reply runner/utils/driver/docker/helpers.go]: At first I thought you were talking about the error messages.
Got it now what you meant

**Pull request merged**